### PR TITLE
Added loUpgradeSpendTier

### PR DESCRIPTION
**Pull request by ryan-rushton**

### CODE REVIEW

File: api/shapes/settings.ts [hunk 1]
```diff
@@ -102,6 +102,9 @@ export interface Settings {
   /** The initial status of assume masterwork in the loadout optimizer. */
   readonly loAssumeMasterwork: boolean;
 
+  /** The optimizers material spend tier, effects armors maximum energy when calcuating sets. */
+  readonly loUpgradeSpendTier: number;
```
Comment (bhollis) [inline api/shapes/settings.ts:106]: Should this be an enum?

Comment (ryan-rushton) [reply api/shapes/settings.ts:106]: Yes, I didn't know we could use enums for api types

Comment (ryan-rushton) [reply api/shapes/settings.ts]: Done

**Review by bhollis — APPROVED** [review]

File: api/shapes/settings.ts [hunk 2]
```diff
@@ -183,6 +194,7 @@ export const defaultSettings: Settings = {
     4244567218, //Strength
   ],
   loAssumeMasterwork: false,
+  loUpgradeSpendTier: 0,
```
Comment (bhollis) [inline api/shapes/settings.ts:197]: Nicer to say it's `UpgradeSpendTier.Nothing`

**Review by bhollis — APPROVED** [review]

**Pull request merged**